pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -568,75 +573,61 @@ func (p *Provisioner) populateDomainAndPort(sc *storagev1.StorageClass) error {
-	if maxSize != "" {
-		quotaEnabled = true
-
-		maxSizeInt64, err = toInt64(maxSize)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse maxSize quota for user %q", p.cephUserName)
-		}
+	additionalConfig, err := additionalConfigSpecFromMap(options.ObjectBucketClaim.Spec.AdditionalConfig)
+	if err != nil {
+		return errors.Wrap(err, "failed to process additionalConfig")
 	}
 
-	objectUser, err := p.adminOpsClient.GetUser(p.clusterInfo.Context, admin.User{ID: p.cephUserName})
+	liveQuota, err := p.adminOpsClient.GetUserQuota(p.clusterInfo.Context, admin.QuotaSpec{UID: p.cephUserName})
 	if err != nil {
 		return errors.Wrapf(err, "failed to fetch user %q", p.cephUserName)
 	}
 
-	// enable or disable quota for user
-	if *objectUser.UserQuota.Enabled != quotaEnabled {
-		err = p.adminOpsClient.SetUserQuota(p.clusterInfo.Context, admin.QuotaSpec{UID: p.cephUserName, Enabled: &quotaEnabled})
-		if err != nil {
-			return errors.Wrapf(err, "failed to set user %q quota enabled=%v for obc", p.cephUserName, quotaEnabled)
-		}
+	// Copy only the fields that are actively managed by the provisioner to
+	// prevent passing back undesirable combinations of fields.  It is
+	// known to be problematic to set both MaxSize and MaxSizeKB.
+	currentQuota := admin.QuotaSpec{
+		Enabled:    liveQuota.Enabled,
+		MaxObjects: liveQuota.MaxObjects,
+		MaxSize:    liveQuota.MaxSize,
 	}
+	targetQuota := currentQuota
+
+	// enable or disable quota for user
+	quotaEnabled := (additionalConfig.maxObjects != nil) || (additionalConfig.maxSize != nil)
+
+	targetQuota.Enabled = &quotaEnabled
 
-	if !quotaEnabled {
-		// no need to process anything else if quotas are disabled
-		return nil
+	if additionalConfig.maxObjects != nil {
+		targetQuota.MaxObjects = additionalConfig.maxObjects
+	} else if currentQuota.MaxObjects != nil && *currentQuota.MaxObjects >= 0 {
+		// if the existing value is already negative, we don't want to change it
+		var objects int64 = -1
+		targetQuota.MaxObjects = &objects
 	}
 
-	if *objectUser.UserQuota.MaxObjects != maxObjectsInt64 {
-		err = p.adminOpsClient.SetUserQuota(p.clusterInfo.Context, admin.QuotaSpec{UID: p.cephUserName, MaxObjects: &maxObjectsInt64})
-		if err != nil {
-			return errors.Wrapf(err, "failed to set MaxObjects=%v to user %q", maxObjectsInt64, p.cephUserName)
-		}
+	if additionalConfig.maxSize != nil {
+		targetQuota.MaxSize = additionalConfig.maxSize
+	} else if currentQuota.MaxSize != nil && *currentQuota.MaxSize >= 0 {
+		// if the existing value is already negative, we don't want to change it
+		var size int64 = -1
+		targetQuota.MaxSize = &size
 	}
 
-	if objectUser.UserQuota.MaxSize != &maxSizeInt64 {
-		err = p.adminOpsClient.SetUserQuota(p.clusterInfo.Context, admin.QuotaSpec{UID: p.cephUserName, MaxSize: &maxSizeInt64})
+	diff := cmp.Diff(currentQuota, targetQuota)
+	if diff != "" {
+		logger.Debugf("Quota for user %q has changed. diff:%s", p.cephUserName, diff)
```

Comment:
Logs should always include the name and namespace of the resource as well (here that is probably more helpful to make it corresponding object store rather than the OBC name) so that multi-cluster installs can be debugged accurately.

I think that's p.clusterInfo.Namespace and p.objectStoreName

Reply:
I agree that the namespace and objecstore should be logged.  I'm honestly having trouble figuring out where that should happen.  I don't think we went to keep annotating the errors with the same information at each level as they bubble up.  The path for error propagation is difficult for me to wrap my head around me because the bucket provisioner is ultimately wrapped up and shipped off to to lib-bucket-provisioner.

Reply:
I've chewed on this for a bit.  I think the ns for the obc also needs to be reported since it might not be in the same ns as the cephobjectstore.  I've wrapped this onto errors from the `setAdditonSettings()` call sites in both `Provision()` & `Grant()`.

As an aside, something has been bothering me about those entry points from lib-bucket-provisioner.  It seems that `Provison()` should be calling `Grant()` under the hood or the rough equivalent as `Provision()` doesn't setup a bucket policy.